packages/library-base/src/commonMain/kotlin/io/realm/internal/ObjectReference.kt

```diff
@@ -0,0 +1,150 @@
+package io.realm.internal
+
+import io.realm.RealmObject
+import io.realm.internal.interop.Callback
+import io.realm.internal.interop.NativePointer
+import io.realm.internal.interop.PropertyInfo
+import io.realm.internal.interop.PropertyKey
+import io.realm.internal.interop.RealmInterop
+import io.realm.internal.schema.ClassMetadata
+import io.realm.notifications.ObjectChange
+import io.realm.notifications.internal.DeletedObjectImpl
+import io.realm.notifications.internal.InitialObjectImpl
+import io.realm.notifications.internal.UpdatedObjectImpl
+import kotlinx.coroutines.channels.ChannelResult
+import kotlinx.coroutines.channels.SendChannel
+import kotlinx.coroutines.flow.Flow
+import kotlin.reflect.KClass
+
+public class ObjectReference<T : RealmObject>(internal val type: KClass<T>) :
+    RealmStateHolder,
+    io.realm.internal.interop.RealmObjectInterop,
+    InternalDeleteable,
+    Observable<ObjectReference<out RealmObject>, ObjectChange<out RealmObject>>,
+    Flowable<ObjectChange<out RealmObject>> {
+
+    public lateinit var owner: RealmReference
```

Comment:
All these properties should just be initialized in the constructor now. We would never have an `ObjectReference` without actually having a managed object as it was the case for all these properties that was also present in the `RealmObject`.

Reply:
I left like that because these properties are being directly initialized on the macos `InstrumentedTests`. I have added a secondary constructor to support this test case.